File: agroguardian_app_ios/agroguardian_app_ios/Models/TelemetryData.swift
//
//  TelemetryData.swift
//  agroguardian_app_ios
//
//  Created by Eren on 8.05.2025.
//


import Foundation

struct TelemetryData:Encodable, Decodable,Identifiable {
    var id: String { timestamp }
    let deviceName: String
    let temperature: Double
    let humidity: Double
    let soilMoisture: Double
    let timestamp: String

    private static let isoFormatter: ISO8601DateFormatter = {
        let formatter = ISO8601DateFormatter()
        return formatter
    }()

    var time: Date {
        TelemetryData.isoFormatter.date(from: timestamp) ?? Date()
    }
}

File: agroguardian_app_android/app/src/main/java/com/example/agroguardianappandroid/data/models/TelemetryData.kt
package com.example.agroguardianappandroid.data.models

import java.text.SimpleDateFormat
import java.util.Date
import java.util.Locale

data class TelemetryData(
    val deviceName: String,
    val temperature: Double,
    val humidity: Double,
    val soilMoisture: Double,
    val timestamp: String
) {
    private val isoFormatter = SimpleDateFormat("yyyy-MM-dd'T'HH:mm:ss'Z'", Locale.getDefault())

    val time: Date
        get() {
            return try {
                isoFormatter.parse(timestamp) ?: Date()
            } catch (e: Exception) {
                Date()
            }
        }
}